### PR TITLE
[openweathermap] Remove daily forecast limitation hint from docs.

### PR DESCRIPTION
**Pull request by maihacke**
Signed-off-by: Simon Sielmann <simon.spielman@gmx.de>

Removed outdated hint for limitations for free users from docs. no code change.

https://github.com/openhab/openhab-addons/issues/9907

### CODE REVIEW

**Review by wborn — APPROVED** [review]
Let's hope they don't make everybody addicted to daily forecasts and then change the pricing model again. :wink:

File: bundles/org.openhab.binding.openweathermap/README.md [hunk 1]
```diff
@@ -23,9 +23,6 @@ If your system language is supported by the OpenWeatherMap API it will be used a
 The second thing `weather-and-forecast` supports the [current weather](https://openweathermap.org/current), [5 day / 3 hour forecast](https://openweathermap.org/forecast5) and optional [16 day / daily forecast](https://openweathermap.org/forecast16) services for a specific location.
 It requires coordinates of the location of your interest.
 You can add as many `weather-and-forecast` things for different locations to your setup as you like to observe.
-**Attention**: The daily forecast is only available for [paid accounts](https://openweathermap.org/price).
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.openweathermap/README.md:26]: I am not sure if this is correct. The One Call API supports daily forecast for free, but this section is explaining the legacy Thing querying a different API endpoint which is still limited. The Thing for the One Call API is described in the section below.

Comment (maihacke) [reply bundles/org.openhab.binding.openweathermap/README.md:26]: According to https://openweathermap.org/price there is no difference for daily forecasts in free and paid accounts.

Comment (cweitkamp) [reply bundles/org.openhab.binding.openweathermap/README.md:26]: I disagree. From my pov "Free" plan includes Daily forecast 7 days provided by the One Call API (v3.0). All higher plans additionally provide Daily forecast 16 days provided by the v2.5 version of the API.

**Comment by cweitkamp** [comment]
Closing this PR now.

**Pull request closed without merging**